### ISSUE TITLE
whereBelongsTo not working

### ISSUE THREAD

**Issue by levu42** [staudenmeir/belongs-to-through#94]
Laravel's `whereBelongsTo` isn't working, because [QueriesRelationships.php checks if the relationship is an instance of `BelongsTo`].(https://github.com/laravel/framework/blob/9448e01699d43761e8bcf3d2ff53d0e3efb37432/src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php#L571)

Is it realistic to change the `BelongsToThrough` class to be a subclass of `BelongsTo` instead of `Relation`?

**Comment by levu42** [comment]
@staudenmeir yes, I totally get that! It would need to do something like `where user_id IN(select id from users where country_id = ?)` using subqueries. Currently, I build these subqueries manually, is there some way to do something similar to `whereBelongsTo` with current code?

It would of course be magic, if I could just put a `BelongsToThrough` into `whereBelongsTo` method at some point, not having to worry about the implementation details, when I write my query logic. Maybe that could be achieved by extracting the necessary parts from `BelongsTo` into a contract that can be implemented both by `BelongsTo` & `BelongsToThrough`, so that `whereBelongsTo` can work with both.

**Comment by staudenmeir** [comment]
You're looking for [`whereHas()`](https://laravel.com/docs/10.x/eloquent-relationships#querying-relationship-existence). The short version for your use case is `whereRelation()`:

```php
$posts = Post::whereRelation('country', 'countries.id', $country->id)->get();
```

The better and faster way would be to (define and) use the inverse relationship:

```php
$posts = $country->posts;
```

**Comment by levu42** [comment]
@staudenmeir oh, thanks! I totally forgot about `whereRelation()` because I got so used to using `whereBelongsTo`. The case I use it for needs it this way round, so the inverse relationship I can't use in that case. Thanks for your help :)
